### DISCUSSION TITLE
Noob Coder - Struggling to pass args at transcode Call

### DISCUSSION THREAD

**Discussion by miles-du** [openai/whisper#1423] [Q&A]
I have a working python script using Whisper to do some voice transcribing. I am doing some refinements.
I want to use:
  --threads THREADS     number of threads used by torch for CPU inference; supercedes MKL_NUM_THREADS/OMP_NUM_THREADS
                        (default: 0)

To set the number of threads used to 4 when the script calls Whisper. I have tried all sorts of approaches and syntax trying to make this work with no luck. My last attempt was to set it up first and append it to the transcribe request:

%Initialize Whisper models
tiny_model = whisper.load_model('tiny')
base_model = whisper.load_model('base')

%Create a DecodingOptions object with the threads property set to 4
options = whisper.DecodingOptions(threads=4)

                result = tiny_model.transcribe('wake_detect.wav', options=options)

This does't work. I'm sure I'm just getting something simple wrong. Please help!

**Comment by phineas-pta** [comment]
not all options go through `.transcribe()`

use `torch.set_num_threads(4)`

**Reply by miles-du** [reply]
Thanks for helping!
Whisper has a lot of args,
  --threads THREADS     number of threads used by torch for CPU inference; supercedes MKL_NUM_THREADS/OMP_NUM_THREADS
                        (default: 0)

Seems strange to use something else.


I tried this but got:

  File "E:\PROJECTS\AI_Projects\Bard_Ai_Voice-assistant\main7.py", line 31, in <module>
    torch.set_num_threads(4)
NameError: name 'torch' is not defined

How do I set this up?

**Reply by miles-du** [reply]
Here are my imports. I don't think the script requires torch? Would like to avoid it.

import boto3
from pydub import AudioSegment
from pydub import playback
import tempfile
import os
import speech_recognition as sr
import whisper
from bardapi import Bard

**Reply by phineas-pta** [reply]
yes you need to `import torch`

whuy do you want to avoid it?

**Reply by miles-du** [reply]
Elegance. Probably my lack of experience, but puzzled why can't just pass in-built arg direct to whisper? Is it because it's a script vs CLI?

**Reply by phineas-pta** [reply]
import a package doesn't make things ugly

> why can't just pass in-built arg direct to whisper?

because that's how `whisper` was written, take a look at source code! (file `whisper/transcribe.py`)

**Reply by miles-du** [reply]
Ok, many thanks and of course I'll take your advice. Cheers.

**Reply by miles-du** [reply]
Worked perfect! Thanks phineas.
